src/realm/object-store/c_api/sync.cpp

```diff
@@ -413,6 +413,163 @@ RLM_API void realm_sync_config_set_resync_mode(realm_sync_config_t* config,
     config->client_resync_mode = ClientResyncMode(mode);
 }
 
+RLM_API realm_flx_sync_subscription_set_t* realm_sync_get_latest_subscription_set(const realm_t* realm)
+{
+    if (realm == nullptr)
+        return nullptr;
+
+    return new realm_flx_sync_subscription_set_t((*realm)->get_latest_subscription_set());
+}
+
+RLM_API realm_flx_sync_subscription_set_t* realm_sync_get_active_subscription_set(const realm_t* realm)
+{
+    if (realm == nullptr)
+        return nullptr;
+
+    return new realm_flx_sync_subscription_set_t((*realm)->get_active_subscription_set());
+}
+
+RLM_API realm_flx_sync_mutable_subscription_set_t*
+realm_sync_get_mutable_subscription_set(const realm_flx_sync_subscription_set_t* subscription_set)
+{
+    if (subscription_set == nullptr)
+        return nullptr;
+    return new realm_flx_sync_mutable_subscription_set_t(subscription_set->make_mutable_copy());
+}
+
+RLM_API void realm_sync_get_state_change_notification_async(const realm_flx_sync_subscription_set_t* subscription_set,
+                                                            realm_flx_sync_subscription_set_state_e notify_when,
+                                                            realm_sync_on_subscription_state_changed callback)
+{
+    if (subscription_set) {
+        auto future_state = subscription_set->get_state_change_notification(SubscriptionSet::State{notify_when});
+        using State = StatusWith<SubscriptionSet::State>;
+        auto f = [callback](State state) -> void {
+            if (callback && state.is_ok()) {
```

Comment:
I think if `!state.is_ok()` you also need to call the callback but with the Error state.